lib/bap/bap.mli

```diff
@@ -632,11 +634,12 @@ module Std : sig
       val flag :
         ?docv:string -> ?doc:string -> name:string -> bool param
 
-      (** Reads a value from a parameter *)
-      type 'a reader = 'a param -> 'a
+      (** A witness that can read configured params *)
```

Comment:
can you also add function:
```
val determined : 'a param -> 'a future
```

This will allow a user to build more complex computations based on parameters, if he likes monadic interface better.